### PR TITLE
Fix Android Chrome input date dimensions

### PR DESCRIPTION
**Pull request by tkrotoff**
Under Android Chrome, input date dimensions are broken:
- too tall because of "margin: 1px 24px 1px 4px" inside the shadow DOM
- width is too small by default

### Before 40px high vs 38px for the other inputs:

before

### After 38px high:

after

I've tested by plugging in my Android device via USB then opened chrome://inspect/#devices (USB debugging must be enabled on the device).
Hence the need for an HTTP server

### CODE REVIEW

File: scss/forms/_form-control.scss [hunk 1]
```diff
@@ -44,12 +44,22 @@
...
   }
 
-  // Add some height to date inputs on iOS
-  // https://github.com/twbs/bootstrap/issues/23307
-  // TODO: we can remove this workaround once https://bugs.webkit.org/show_bug.cgi?id=198959 is resolved
   &::-webkit-date-and-time-value {
+    // Android Chrome + form-control's "width: 100%" makes the input width too small
+    // "2022/04/10" = 10 characters
+    // Tested under Android 11 / Chrome 89 & Android 12 / Chrome 100
+    min-width: 10ch;
```
Comment (tkrotoff) [inline scss/forms/_form-control.scss:51]: Not sure min-width vs width

File: package.json [hunk 2]
```diff
@@ -81,8 +81,9 @@
     "docs-compile": "npm run docs-build",
     "docs-vnu": "node build/vnu-jar.js",
     "docs-lint": "npm run docs-vnu",
-    "docs-serve": "hugo server --port 9001 --disableFastRender",
-    "docs-serve-only": "npx sirv-cli _site --port 9001",
+    "docs-serve": "hugo server --port 9001 --bind=0.0.0.0 --disableFastRender",
+    "docs-serve-only": "npx http-server _site --port 9001",
+    "visual-tests-serve": "npx http-server . --port 9001 -o js/tests/visual",
```
Comment (tkrotoff) [inline package.json:86]: http-server has nice features out of the box compared to sirv-cli:

- `-o` option: https://github.com/http-party/http-server#available-options

> Open browser window after starting the server. Optionally provide a URL path to open. e.g.: -o /other/dir/

- it lists the files in directories (no need to write an index.html)

image

Comment (XhmikosR) [reply package.json:86]: We try to keep our deps to a minimum. And that's the reason we use sirv-cli and we plan to do so.

**Review by XhmikosR — DISMISSED** [review]
See comments

File: package.json (hunk 2 above)
Comment (XhmikosR) [reply package.json:86]: We don't plan to change to http-serve, sorry. Keep your PRs clean, one change at a time. Similarly, we don't want to change to `0.0.0.0`.

Comment (tkrotoff) [reply package.json]: I've reverted the 2 commits.

Still interested to know what you plan to do with the visual tests.
If polished they can be a great asset and catch lot of problems earlier.

I'm using "visual tests" for 2 libraries based on Bootstrap:
- https://codesandbox.io/s/github/tkrotoff/bootstrap-floating-label/tree/codesandbox.io (https://github.com/tkrotoff/bootstrap-floating-label)
- https://codesandbox.io/s/github/tkrotoff/bootstrap-input-spinner/tree/codesandbox.io (https://github.com/tkrotoff/bootstrap-input-spinner)

They quickly became essential for me for non-regressions and edge cases.

image

**Comment by tkrotoff** [comment]
Any news?

**Comment by tkrotoff** [comment]
Any news?

**Comment by mdo** [comment]
Missed this myself. Is the width bug present in any other browser? I don’t have an Android to test myself.

**Comment by tkrotoff** [comment]
Visual tests js/tests/visual/input.html

On macOS Monterey 12.6.2:
- Chrome 109: no problem
- Firefox 108: no problem
- Safari 16.2: no problem
- Edge 109: no problem

On Android 13 (Pixel 4):
- Chrome 109: **width & height problem**
- Firefox 109: no problem

On iOS 16.2 (iPhone 14 Pro Max Simulator):
- Safari 16.2: no height problem (thx to https://github.com/twbs/bootstrap/blob/v5.3.0-alpha1/scss/forms/_form-control.scss#L47-L53), **width problem**

iOS 16.2 Safari 16.2

The width problem on Safari iOS is also fixed by this PR:

iOS 16.2 Safari 16.2 fix `min-width: 10ch`

File: .bundlewatch.config.json [hunk 3]
```diff
@@ -26,7 +26,7 @@
     },
     {
       "path": "./dist/css/bootstrap.css",
-      "maxSize": "32.25 kB"
+      "maxSize": "32.26 kB"
```
Comment (mdo) [inline .bundlewatch.config.json:29]:
```suggestion
      "maxSize": "32.5 kB"
```

**Review by mdo — APPROVED** [review]

**Comment by mdo** [comment]
Finally! 😅

**Pull request merged**